### PR TITLE
Allow for string values to be used in scale.getPixelForValue

### PR DESCRIPTION
**Pull request by ground5hark**
Currently the plugin does not allow for labels using non-numerical values. `getPixelForValue` supports string values along the axis, so the plugin should probably have this flexibility too.

For example, our team uses timestamps (HH:mm) along the y-axis.

### CODE REVIEW

**Comment by ground5hark** [comment]
To help illustrate this, here's the non-working (master) version:
https://codepen.io/ground5hark/pen/NjBYPq

Here it is on the working (forked) version:
https://codepen.io/ground5hark/pen/xdJWxY

@compwright mind taking a quick peek at this?

File: chartjs-plugin-annotation.js [hunk 1]
```diff
@@ -277,7 +277,11 @@ function objectId() {
 }
 
 function isValid(num) {
-	return !isNaN(num) && isFinite(num);
+  if (!isNaN(num)) {
```
Comment (compwright) [inline chartjs-plugin-annotation.js]: Please move this change to src/helpers.js

File: chartjs-plugin-annotation.js [hunk 2]
```diff
@@ -849,25 +853,25 @@ module.exports = function(Chart) {
 				ya = padHeight + view.labelYAdjust;
 				xa = (width / 2) + view.labelXAdjust;
 				ret.y = view.y1 + ya;
-				ret.x = (isFinite(line.m) ? line.getY(ret.y) : view.x1) - xa;
+				ret.x = (isFinite(line.m) ? line.getX(ret.y) : view.x1) - xa;
```
Comment (compwright) [inline chartjs-plugin-annotation.js]: Please move this change to src/types/line.js

File: chartjs-plugin-annotation.js [hunk 3]
```diff
@@ -849,25 +853,25 @@ module.exports = function(Chart) {
...
+
 			// left align
 			case view.mode == horizontalKeyword && view.labelPosition == "left":
 				xa = padWidth + view.labelXAdjust;
 				ya = -(height / 2) + view.labelYAdjust;
 				ret.x = view.x1 + xa;
 				ret.y = line.getY(ret.x) + ya;
 			break;
-			
+
```
Comment (compwright) [inline chartjs-plugin-annotation.js]: Please revert all changes to chartjs-plugin-annotation.js and chartjs-plugin-annotation.min.js. We will compile these when we make a release.

**Comment by compwright** [comment]
Duplicate of #50

**Pull request closed without merging**